app/controllers/users_controller.rb

```diff
@@ -1490,28 +1498,34 @@ def confirm_admin
     end
   end
 
+  def confirm_session
```

Comment:
Just curious, why do we need to introduce this endpoint in the context of passkey which is passwordless?

Reply:
Good question. I think an extra level of security for the first time you add a passkey. Similar to 2FA methods (where we ask for a password) or changing your password (where we confirm via email), we want to confirm the user is who they say they are before allowing them to add a new passkey. 

See also https://github.com/discourse/discourse/pull/23587#discussion_r1353182689